pkg/schema/v1/replica_set.go

```diff
@@ -119,6 +124,47 @@ func (r *ReplicaSet) Obtain(k8s kmetav1.Object) {
 	}
 }
 
+func (r *ReplicaSet) getIcingaState() (IcingaState, string) {
+	if r.DesiredReplicas < 1 {
+		reason := fmt.Sprintf("ReplicaSet %s/%s has an invalid desired replica count: %d", r.Namespace, r.Name, r.DesiredReplicas)
+		return Unknown, reason
+	}
+
+	// Check if the ReplicaSet was created within the last 5 minutes
```

Comment:
I would use a function for the grace period detection and make all other relevant PRs dependent on this one:

```
// IsWithinGracePeriod ...
func IsWithinGracePeriod(k kmetav1.Object) *string {
	const gracePeriod = 5 * time.Minute

	deadline := k.GetCreationTimestamp().Add(gracePeriod)
	now := time.Now()
	if now.Before(deadline) {
		key, _ := kcache.MetaNamespaceKeyFunc(k)
		reason := fmt.Sprintf(
			"%s %s is within grace period until %s, so its state is not yet evaluated.",
			types.Name(k),
			key,
			deadline)

		return &reason
	}

	return nil
}
```

I decided to add the deadline date instead of the time left, because the time left would imply that the state is evaluated directly after the time has expired, but this is not the case, as it is only evaluated when Kubernetes updates the resource.